src/ui/public/vis/agg_configs.js

```diff
@@ -145,14 +144,45 @@ export default function AggConfigsFactory(Private) {
     });
 
     removeParentAggs(dslTopLvl);
-
     return dslTopLvl;
   };
 
   AggConfigs.prototype.getRequestAggs = function () {
-    return _.sortBy(this, function (agg) {
+
+
+    const aggregations = _.sortBy(this, function (agg) {
       return agg.schema.group === 'metrics' ? 1 : 0;
     });
+
+
+    let shouldAddGeocentroid = false;
```

Comment:
You think we can use https://github.com/elastic/kibana/blob/master/src/ui/public/agg_types/agg_type.js#L129?